fs/fs.go

```diff
@@ -642,6 +648,12 @@ type PutStreamer interface {
 	PutStream(in io.Reader, src ObjectInfo, options ...OpenOption) (Object, error)
 }
 
+// PublicLinker is an optional interface for Fs
+type PublicLinker interface {
+	// PublicLink generates a public link to the remote path (usually readable by anyone)
+	PublicLink(fileName string) (string, error)
```

Comment:
This `fileName` should probably be `remote` and be relative to the Fs root as per normal rclone usage.